.github/workflows/compliance.yml

```
@@ -1,18 +1,23 @@
 name: Compliance Checks
 
 on:
-  pull_request:
-    types:
-    - edited
-    - opened
-    - reopened
-    - synchronize
+  workflow_call:
```

I understand the motivation to move the twister job from `pull_request_target` to `pull_request`, but what is the motivation behind changing this?

---

https://github.com/zephyrproject-rtos/infrastructure/issues/21

---

Sure, but that is a very wide proposal. This PR implements something partially different, and I was wondering whether we could simplify things by just having the compliance workflow run on `pull_request` and then have the twister workflow be triggered on `workflow_run`. To me this sounds like a simpler solution and equally safe, since the only thing that would run automatically is on `pull_request` with no write access.

---

> This PR implements something partially different,

how is this different? The ultimate goal is to have a super-workflow with depenecies and a way to fail fast, which this PR does. see here https://github.com/zephyrproject-rtos/zephyr-testing/actions/runs/13906350118 for how this looks like

This PR uses existing twister workflow to start compliance, you are asking for compliance to start twister. I am sure I have tried this last year and it did not work or something was missing.
There are lots of special cases that are different from the doc workflow, i.e. we might fail compliance but we still want to run twister, twister being skipped because nothing has changed and other things.
I can spend some time going backwards and doing all the steps to show what does not work, which is not fun and will take time.

---

All I am asking is that we document why we did it in this (in my  opinion) counter-intuitive way. If we are to go back in time later and look at how this happened, for me this PR and its commits does not provide a clear explanation of why. I don't typically block PRs but with the amount time I've spent on GH actions I've learned to document things very clearly for our future selves :)